plug.vim

```diff
@@ -1206,11 +1206,11 @@ function! s:spawn(name, cmd, opts)
             \ 'new': get(a:opts, 'new', 0) }
   let s:jobs[a:name] = job
   let cmd = has_key(a:opts, 'dir') ? s:with_cd(a:cmd, a:opts.dir) : a:cmd
+  let argv = add(s:is_win ? ['cmd', '/c'] : ['sh', '-c'], cmd)
   if !empty(job.batchfile)
     call writefile(["@echo off\r", cmd . "\r"], job.batchfile)
-    let cmd = s:shellesc(expand(job.batchfile))
+    let argv = s:vim8 ? 'cmd.exe '.s:shellesc(job.batchfile) : ['cmd.exe', job.batchfile]
```

Comment:
Should have done this from the start but without the batchfile for jobs.

Reply:
In `line: 1230`, the `argv` becomes:
```
cmd.exe ^"C:/Users/mike/AppData/Local/Temp/VIE923.tmp.bat^"
```

It seems `/c` is missed:
```vim
let argv = s:vim8 ? 'cmd.exe '.s:shellesc(job.batchfile) : ['cmd.exe', job.batchfile]
```

Reply:
That's intentional. Did it work?

Reply:
No, it will shown as:
```
Updating plugins (0/1)
[ ]

* delimitMate: D:\Code>
```

The position is the current PWD.

Reply:
Slightly better. It failed immediately before the last commit.

I need to debug this manually. Can you provide a minimal vimrc with only your shell options?

Reply:
```vim
set shell=C:/bin/cygwin64/bin/bash
set shellslash
set shellpipe=\|&\ tee
set shellquote=
set shellredir=&>
set shellcmdflag=-c

set shellxquote=\"
set shellxescape=  
```

Reply:
I assume that `shellslash` affecting a function that outputs a path with the wrong pathsep is the culprit. vim-plug need the same set/unset shellslash hack from fzf's Vim plugin.

@mikelue Do you pass any Windows filepath to vim-plug like `plug#begin`?

Reply:
I tested with my vimrc. It works with sh.exe with `noshellslash`.

Reply:
It works with `shellslash` now.

Reply:
> I assume that `shellslash` affecting a function that outputs a path with the wrong pathsep is the culprit. vim-plug need the same set/unset shellslash hack from fzf's Vim plugin.
> 
> @mikelue Do you pass any Windows filepath to vim-plug like `plug#begin`?

Yes, I use `call plug#begin("$HOME/vimfiles/plugged")`.
The `$HOME` is same as `$USERPROFILE` in windows system.

Reply:
> The $HOME is same as $USERPROFILE in windows system.

Vim prefers `%HOMEDRIVE%%HOMEPATH%` but it's fine to assume this on a regular user account. Anyway, vim-plug will expand this to what Vim thinks `$HOME` should be if unset on Windows.